File: 0226-invert-binary-tree/0226-invert-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        // 예외 처리
        if(root == null) {
            return null;
        }
        
        // 반복 DFS를 위한 스택 선언
        Queue<TreeNode> queue = new LinkedList<>();
        
        // 스택에 루트부터 삽입
        queue.add(root);
        
        while(!queue.isEmpty()) {
            TreeNode node = queue.poll();
            
            // 왼쪽/오른쪽 노드 스왑
            TreeNode tmp = node.left;
            node.left = node.right;
            node.right = tmp;
            
            // 스택에 왼쪽 노드 추가
            if(node.left != null) {
                queue.add(node.left);
            }
            
            // 스택에 오른쪽 노드 추가
            if(node.right != null) {
                queue.add(node.right);
            }
        }
        
        return root;
    }
}

File: 0226-invert-binary-tree/0226-invert-binary-tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun invertTree(root: TreeNode?): TreeNode? {
        // 빈 노드 예외 처리
        if(root == null) {
            return root
        }
        
        // 왼쪽/오른쪽 자식 노드 스왑
        root.left = root.right.also {root.right = root.left}
        
        invertTree(root.left)   // 왼쪽 자식 노드 dfs
        invertTree(root.right)  // 오른쪽 자식 노드 dfs
        
        return root
    }
}